msal/src/Microsoft.Identity.Client/Features/ConfidentialClient/ConfidentialClientApplication.cs

```diff
@@ -150,7 +150,42 @@ public sealed class ConfidentialClientApplication : ClientApplicationBase, IConf
         {
             return
                 await
-                    AcquireTokenForClientCommonAsync(scopes, forceRefresh, ApiEvent.ApiIds.AcquireTokenForClientWithScopeRefresh).ConfigureAwait(false);
+                    AcquireTokenForClientCommonAsync(scopes, forceRefresh, ApiEvent.ApiIds.AcquireTokenForClientWithScopeRefresh, false).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Acquires token from the service for the confidential client. This method attempts to look up valid access token in the cache.
+        /// </summary>
+        /// <param name="scopes">Array of scopes requested for resource</param>
+        /// <param name="sendCert">This parameter enables application developers to achieve easy certificates roll-over
+        /// in Azure AD: setting this parameter to true will send the public certificate to Azure AD
+        /// along with the token request, so that Azure AD can use it to validate the subject name based on a trusted issuer policy.
+        /// This saves the application admin from the need to explicitly manage the certificate rollover
+        /// (either via portal or powershell/CLI operation)</param>
+        /// <returns>Authentication result containing application token for the requested scopes</returns>
+        public async Task<AuthenticationResult> AcquireTokenForClientAsync(bool sendCert, IEnumerable<string> scopes)
```

Comment:
why is sendCert the first parameter?

Reply:
Visual studio showed an error when I put is behind the scopes parameter because there is already an overload that takes form of AcquireTokenForClientAsync(IEnumerable<string>, bool).

Thus there would be ambiguity between:
AcquireTokenForClientAsync(IEnumerable<string> scopes, bool **sendCertificate**)
and
AcquireTokenForClientAsync(IEnumerable<string> scopes, bool **forceRefresh**)

Reply:
@jmprieur
Perhaps it should be AquireTokenWithCertificateAsync ?

---
In reply to: [206953484](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1124#discussion_r206953484) [](ancestors = 206953484)

Reply:
@trawalke pls. follow up with Jean-Marc on this.

---
In reply to: [206999250](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1124#discussion_r206999250) [](ancestors = 206999250,206953484)